zk_test.go

```diff
@@ -161,28 +227,85 @@ func TestIntegration_CreateContainer(t *testing.T) {
+	tests := []struct {
+		name          string
+		createFlags   int32
+		specifiedPath string
+		wantErr       string
+	}{
+		{
+			name:        "valid create container",
+			createFlags: FlagContainer,
+		},
+		{
+			name:        "valid create container hard coded flag int",
+			createFlags: 4,
+			// container flag, ensure matches ZK Create Mode (change detector test)
+		},
+		{
+			name:          "invalid path",
+			createFlags:   FlagContainer,
+			specifiedPath: "not/valid",
+			wantErr:       "zk: invalid path",
+		},
+		{
+			name:        "invalid create mode",
+			createFlags: 999,
+			wantErr:     "invalid flag value: [999]",
+		},
+		{
+			name:        "invalid containers cannot be persistant",
+			createFlags: FlagPersistent,
+			wantErr:     ErrInvalidFlags.Error(),
+		},
+		{
+			name:        "invalid containers cannot be ephemeral",
+			createFlags: FlagEphemeral,
+			wantErr:     ErrInvalidFlags.Error(),
+		},
+		{
+			name:        "invalid containers cannot be sequential",
+			createFlags: FlagSequence,
+			wantErr:     ErrInvalidFlags.Error(),
+		},
+		{
+			name:        "invalid container and sequential",
+			createFlags: FlagContainer | FlagSequence,
+			wantErr:     ErrInvalidFlags.Error(),
+		},
+		{
+			name:        "invliad TTLs cannot be used with Container znodes",
+			createFlags: FlagTTL,
+			wantErr:     ErrInvalidFlags.Error(),
+		},
+	}
+
+	const testPath = "/container_test_znode"
+	// create sub node to create per test in avoiding using the root path.
+	_, err = zk.Create(testPath, nil /* data */, FlagPersistent, WorldACL(PermAll))
+	requireNoErrorf(t, err)
 
-	if err := zk.Delete(path, -1); err != nil && err != ErrNoNode {
-		t.Fatalf("Delete returned error: %+v", err)
-	}
-	if _, err := zk.CreateContainer("", []byte{1, 2, 3, 4}, FlagTTL, WorldACL(PermAll)); err != ErrInvalidPath {
-		t.Fatalf("Create path check failed")
-	}
-	if _, err := zk.CreateContainer(path, []byte{1, 2, 3, 4}, 0, WorldACL(PermAll)); err != ErrInvalidFlags {
-		t.Fatalf("Create flags check failed")
-	}
-	if p, err := zk.CreateContainer(path, []byte{1, 2, 3, 4}, FlagTTL, WorldACL(PermAll)); err != nil {
-		t.Fatalf("Create returned error: %+v", err)
-	} else if p != path {
-		t.Fatalf("Create returned different path '%s' != '%s'", p, path)
-	}
-	if data, stat, err := zk.Get(path); err != nil {
-		t.Fatalf("Get returned error: %+v", err)
-	} else if stat == nil {
-		t.Fatal("Get returned nil stat")
-	} else if len(data) < 4 {
-		t.Fatal("Get returned wrong size data")
+	for idx, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			path := filepath.Join(testPath, fmt.Sprint(idx))
+			if tt.specifiedPath != "" {
+				path = tt.specifiedPath
+			}
+
+			_, err := zk.CreateContainer(path, []byte{12}, tt.createFlags, WorldACL(PermAll))
+			if tt.wantErr == "" {
+				requireNoErrorf(t, err, fmt.Sprintf("error not expected: path; %q; flags %v", path, tt.createFlags))
+				return
+			}
+
+			// want an error
+			if err == nil {
```

Comment:
test assertion helpers would be great :D maybe a follow up issue/pr

Reply:
I have kept the repo at no external deps. soon or im thinking to limit to test deps only. it may help a lot moving forward. but i do respect and want to keep the lib code void of a tree of third party code, it makes things so much nicer to maintain with no downstream library behaviors

Reply:
Yeah sucks that go.mod doesn't support test dependencies. Agree that having 0 deps is really nice. Maybe we could write some minimal set of common test utils on our own.